File: frontend/src/services/dashboardService.ts
import apiClient from './apiClient';

export interface DashboardConfig {
  id: number;
  userId: number;
  dashboardIds: string[];
}

class DashboardService {
  async getAvailableDashboards(): Promise<string[]> {
    const response = await apiClient.get<string[]>('/dashboards/available');
    return response.data;
  }

  async getUserDashboardConfig(): Promise<DashboardConfig> {
    const response = await apiClient.get<DashboardConfig>('/dashboards/me');
    return response.data;
  }

  async saveDashboardConfig(dashboardIds: string[]): Promise<DashboardConfig> {
    const response = await apiClient.put<DashboardConfig>('/dashboards/me', dashboardIds);
    return response.data;
  }
}

export default new DashboardService();

File: backend/src/main/java/com/smartsolutions/hub/service/DashboardService.java
package com.smartsolutions.hub.service;

import com.smartsolutions.hub.dto.DashboardConfigDTO;
import com.smartsolutions.hub.model.DashboardConfig;
import com.smartsolutions.hub.repository.DashboardConfigRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class DashboardService {
    
    private final DashboardConfigRepository dashboardConfigRepository;
    
    // Available dashboards for all users
    private static final List<String> AVAILABLE_DASHBOARDS = List.of(
        "kpi-overview",
        "safety-metrics",
        "project-timeline",
        "resource-allocation",
        "budget-tracking",
        "quality-control"
    );
    
    public List<String> getAvailableDashboards() {
        return AVAILABLE_DASHBOARDS;
    }
    
    public DashboardConfigDTO getUserDashboardConfig(Long userId) {
        DashboardConfig config = dashboardConfigRepository.findByUserId(userId)
            .orElseGet(() -> {
                DashboardConfig newConfig = new DashboardConfig();
                newConfig.setUserId(userId);
                newConfig.setDashboardIds(new ArrayList<>(List.of("kpi-overview", "safety-metrics")));
                return dashboardConfigRepository.save(newConfig);
            });
        
        return toDTO(config);
    }
    
    @Transactional
    public DashboardConfigDTO saveDashboardConfig(Long userId, List<String> dashboardIds) {
        DashboardConfig config = dashboardConfigRepository.findByUserId(userId)
            .orElseGet(() -> {
                DashboardConfig newConfig = new DashboardConfig();
                newConfig.setUserId(userId);
                return newConfig;
            });
        
        config.setDashboardIds(new ArrayList<>(dashboardIds));
        DashboardConfig saved = dashboardConfigRepository.save(config);
        return toDTO(saved);
    }
    
    private DashboardConfigDTO toDTO(DashboardConfig config) {
        return new DashboardConfigDTO(
            config.getId(),
            config.getUserId(),
            config.getDashboardIds()
        );
    }
}
